cmd/internal/cli/build_linux.go

```diff
@@ -73,9 +78,12 @@ func fakerootExec(isDeffile bool) {
 
 	var err error
 	uid := uint32(os.Getuid())
-	if uid != 0 && !fakeroot.IsUIDMapped(uid) {
+	if buildArgs.ignoreUserns {
+		os.Setenv("APPTAINER_IGNORE_USERNS", "1")
+	} else if (uid != 0 && !fakeroot.IsUIDMapped(uid)) || buildArgs.userns {
```

Comment:
The ignoreUserns and userns options aren't relevant here.  The relevant one is ignoreSubuid.  The if should be
```
    if uid != 0 && (!fakeroot.IsUIDMapped(uid) || buildArgs.ignoreSubuid) {
```